make test fails on db test

Here is the output when I do 'make test' (long output; after this, at the end I include backtrace output from lldb on the resulting core dumps). I use a Mac running MacOS 12.2 and checked out the main acton branch May 3 at 14:45.

```
acton> make test
/Library/Developer/CommandLineTools/usr/bin/make -C backend test
make[1]: Nothing to be done for `test'.
/Library/Developer/CommandLineTools/usr/bin/make -C test
/Library/Developer/CommandLineTools/usr/bin/make actonc/project/simple actonc/project/missing_src actonc/project/auto_root env/listen_err rts/argv1 rts/argv2 rts/argv3 rts/exit_0 rts/exit_1 test_db_app test_db_app_no_quorum test_db_app_recovery test_db_app_resume_tcp_client test_db_app_resume_tcp_server test_acton_rts_sleep test_random test_time rts_sleep regression stdlib/test_numpy
cd actonc/project/simple && /Users/sydow/acton/dist/bin/actonc --cpedantic build
actonc/project/simple/out/types/a.h
actonc/project/simple/out/types/a.ty
actonc/project/simple/out/types/b.h
actonc/project/simple/out/types/b.ty
cd actonc/project/missing_src && /Users/sydow/acton/dist/bin/actonc --cpedantic build; EXIT_CODE=$?; if [ $EXIT_CODE -ne 1 ]; then exit 1; fi
actonc: Missing src/ directory
cd actonc/project/missing_src && /Users/sydow/acton/dist/bin/actonc --cpedantic build 2>&1 | grep "Missing src/ directory"
actonc: Missing src/ directory
cd actonc/project/auto_root && /Users/sydow/acton/dist/bin/actonc --cpedantic build
ls: actonc/project/auto_root/out/rel/bin/test: No such file or directory
make[2]: [actonc/project/auto_root] Error 1 (ignored)
/Users/sydow/acton/dist/bin/actonc --cpedantic --root main env/listen_err.act
Compiling listen_err.act for release
./env/listen_err
This is the second error handler, which we expect to be called because we cannot bind... now exiting
/Users/sydow/acton/dist/bin/actonc --cpedantic --root main rts/argv1.act
Compiling argv1.act for release
./rts/argv1 --rts-verbose foo --bar --rts-verbose
14:32:50.433810 INFO  rts/rts.c           : 1853: RTS   : Detected 8 CPUs: Using 8 worker threads for 1:1 mapping with CPU affinity set.

Expected: [foo, --bar]
env.argv: [foo, --bar]
All is well
./rts/argv1 --rts-verbose --rts-wthreads 7 foo --bar
14:32:50.438659 INFO  rts/rts.c           : 1856: RTS   : Detected 8 CPUs: Using 7 worker threads. No CPU affinity used.

Expected: [foo, --bar]
env.argv: [foo, --bar]
All is well
./rts/argv1 --rts-verbose --rts-wthreads=7 foo --bar
14:32:50.442777 INFO  rts/rts.c           : 1856: RTS   : Detected 8 CPUs: Using 7 worker threads. No CPU affinity used.

Expected: [foo, --bar]
env.argv: [foo, --bar]
All is well
./rts/argv1 --rts-verbose --rts-wthreads 7 foo --bar 2>&1 | grep "Using 7 worker threads"
14:32:50.450175 INFO  rts/rts.c           : 1856: RTS   : Detected 8 CPUs: Using 7 worker threads. No CPU affinity used.
./rts/argv1 --rts-verbose --rts-wthreads=7 foo --bar 2>&1 | grep "Using 7 worker threads"
14:32:50.457864 INFO  rts/rts.c           : 1856: RTS   : Detected 8 CPUs: Using 7 worker threads. No CPU affinity used.
/Users/sydow/acton/dist/bin/actonc --cpedantic --root main rts/argv2.act
Compiling argv2.act for release
./rts/argv2 --rts-verbose --rts-wthreads=7 -- foo --bar --rts-verbose
14:32:50.945464 INFO  rts/rts.c           : 1856: RTS   : Detected 8 CPUs: Using 7 worker threads. No CPU affinity used.

Expected: [--, foo, --bar, --rts-verbose]
env.argv: [--, foo, --bar, --rts-verbose]
All is well
/Users/sydow/acton/dist/bin/actonc --cpedantic --root main rts/argv3.act
Compiling argv3.act for release
./rts/argv3 --rts-wthreads 2>&1 | grep "ERROR: --rts-wthreads requires an argument."
ERROR: --rts-wthreads requires an argument.
Test success, saw expected error message
/Users/sydow/acton/dist/bin/actonc --cpedantic --root main rts/exit_0.act
Compiling exit_0.act for release
./rts/exit_0
/Users/sydow/acton/dist/bin/actonc --cpedantic --root main rts/exit_1.act
Compiling exit_1.act for release
./rts/exit_1; EXIT_CODE=$?; if [ $EXIT_CODE -ne 1 ]; then exit 1; fi
/Users/sydow/acton/dist/bin/actonc --cpedantic --root main test_db_app.act
Compiling test_db_app.act for release
./test_db.py TestDbApps.test_app
E
======================================================================
ERROR: test_app (__main__.TestDbApps)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 45, in mon_cmd
    colpos = buf.index(b':')
ValueError: subsection not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 50, in mon_cmd
    raise ConnectionError("remote hung up")
ConnectionError: remote hung up

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 41, in mon_cmd
    sock.connect(address)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 41, in mon_cmd
    sock.connect(address)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 41, in mon_cmd
    sock.connect(address)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 41, in mon_cmd
    sock.connect(address)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 41, in mon_cmd
    sock.connect(address)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sydow/acton/test/./test_db.py", line 345, in setUp
    self.dbc.start()
  File "/Users/sydow/acton/test/./test_db.py", line 191, in start
    mbs = db.get_vc()
  File "/Users/sydow/acton/test/./test_db.py", line 142, in get_vc
    data = self.get_membership()
  File "/Users/sydow/acton/test/./test_db.py", line 150, in get_membership
    return mon_cmd(self.mon_sock, "membership", retries=5)
  File "/Users/sydow/acton/test/./test_db.py", line 71, in mon_cmd
    return mon_cmd(address, cmd, retries-1)
  File "/Users/sydow/acton/test/./test_db.py", line 71, in mon_cmd
    return mon_cmd(address, cmd, retries-1)
  File "/Users/sydow/acton/test/./test_db.py", line 71, in mon_cmd
    return mon_cmd(address, cmd, retries-1)
  [Previous line repeated 2 more times]
  File "/Users/sydow/acton/test/./test_db.py", line 73, in mon_cmd
    raise MonIntermittentError("Unable to get data from remote")
MonIntermittentError: Unable to get data from remote

----------------------------------------------------------------------
Ran 1 test in 37.250s

FAILED (errors=1)
make[2]: *** [test_db_app] Error 1
make[1]: *** [test] Error 2
make: *** [test] Error 2
acton> 
```

The run produced two core dumps. Here are the backtraces from lldb (they are identical)

```
~> lldb -c /cores/core.68571 
(lldb) target create --core "/cores/core.68571"
Core file '/cores/core.68571' (x86_64) was loaded.
(lldb) bt all
* thread #1, stop reason = signal SIGSTOP
  * frame #0: 0x0000000105fef8f7 actondb`__darwin_fd_isset + 23
    frame #1: 0x0000000105feeb2c actondb`main + 5532
    frame #2: 0x000000011140c4fe dyld`start + 462
(lldb) 
```

```
~> lldb -c /cores/core.68572
(lldb) target create --core "/cores/core.68572"
Core file '/cores/core.68572' (x86_64) was loaded.
(lldb) bt all
* thread #1, stop reason = signal SIGSTOP
  * frame #0: 0x0000000106ee58f7 actondb`__darwin_fd_isset + 23
    frame #1: 0x0000000106ee4b2c actondb`main + 5532
    frame #2: 0x00000001153a14fe dyld`start + 462
(lldb) 
```

--- a/backend/actondb.c
+++ b/backend/actondb.c
@@ -2508,7 +2508,7 @@ int main(int argc, char **argv) {
 			log_debug("select returned %d/%d!", status, errno);
 
         // Monitor socket
-        if (FD_ISSET(mon_sock, &readfds)) {
+        if (mon_sock > 0 && FD_ISSET(mon_sock, &readfds)) {
             if (mon_client_sock == -1) {
                 socklen_t t = sizeof(mon_client_sock);
                 if ((mon_client_sock = accept(mon_sock, (struct sockaddr *)&mon_client_addr, &t)) == -1) {
@@ -2519,7 +2519,7 @@ int main(int argc, char **argv) {
         }
 
         // Monitor socket client requests
-        if (mon_client_sock && FD_ISSET(mon_client_sock, &readfds)) {
+        if (mon_client_sock > 0 && FD_ISSET(mon_client_sock, &readfds)) {
             char *buf_base, *str;
             size_t len;
             ssize_t bytes_read = recv(mon_client_sock, &mon_rbuf[mon_buf_used], sizeof(mon_rbuf) - mon_buf_used, 0);

